src/main/java/edu/kis/powp/jobs2d/command/ComplexCommand.java

```diff
@@ -27,4 +27,12 @@ public void execute(Job2dDriver job2dDriver) {
 		}
 	}
 
+	@Override
+	public ComplexCommand clone() throws CloneNotSupportedException {
+		ComplexCommand tempCommand = new ComplexCommand();
+		for(DriverCommand driverCommand : this.commands) {
+			tempCommand.appendCommand(driverCommand);
```

Comment:
For copy to be "deep", you need to apply it recursive, i.e., make a copy of subcommands